bofire/surrogates/surrogate.py

```diff
@@ -44,27 +45,80 @@ def predict(self, X: pd.DataFrame) -> pd.DataFrame:
             Xt[c] = pd.to_numeric(Xt[c], errors="raise")
         # predict
         preds, stds = self._predict(Xt)
+        # set up column names
+        pred_cols = []
+        sd_cols = []
+        for featkey in self.outputs.get_keys():
+            if hasattr(self.outputs.get_by_key(featkey), "categories"):
```

Comment:
@jduerholt, there is a type error failing here. How do you recommend checking if the output type is categorical? Line 71 below also fails...

Reply:
`for feat in self.outputs.get(CategoricalOutput)`: Doing it like this, you only get categorical output features ;) The same filtereing also applies to `get_keys`. Have a look at the method in `domain.features.py`, they are super helpful and I use them really a lot, so you should familiarize with them.

Reply:
If you only need the keys use `get_keys(CategoricalOutput)`, it could be that you have still a type error, then add a `type: ignore`